server/src/main/java/com/bfwg/config/WebSecurityConfig.java

```diff
@@ -30,60 +27,59 @@
 @EnableGlobalMethodSecurity(prePostEnabled = true)
 public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
 
-  @Value("${jwt.cookie}")
-  private String TOKEN_COOKIE;
-
-  @Bean
-  public TokenAuthenticationFilter jwtAuthenticationTokenFilter() throws Exception {
-    return new TokenAuthenticationFilter();
-  }
-
-  @Bean
-  @Override
-  public AuthenticationManager authenticationManagerBean() throws Exception {
-    return super.authenticationManagerBean();
-  }
-
-  @Bean
-  public PasswordEncoder passwordEncoder() {
-    return new BCryptPasswordEncoder();
-  }
-
-  @Autowired
-  private CustomUserDetailsService jwtUserDetailsService;
-
-  @Autowired
-  private RestAuthenticationEntryPoint restAuthenticationEntryPoint;
-
-  @Autowired
-  private LogoutSuccess logoutSuccess;
-
-  @Autowired
-  public void configureGlobal(AuthenticationManagerBuilder authenticationManagerBuilder)
-      throws Exception {
-    authenticationManagerBuilder.userDetailsService(jwtUserDetailsService)
-        .passwordEncoder(passwordEncoder());
-
-  }
-
-  @Autowired
-  private AuthenticationSuccessHandler authenticationSuccessHandler;
-
-  @Autowired
-  private AuthenticationFailureHandler authenticationFailureHandler;
-
-  @Override
-  protected void configure(HttpSecurity http) throws Exception {
-    http.csrf().ignoringAntMatchers("/api/login", "/api/signup")
-        .csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse()).and()
-        .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS).and()
-        .exceptionHandling().authenticationEntryPoint(restAuthenticationEntryPoint).and()
-        .addFilterBefore(jwtAuthenticationTokenFilter(), BasicAuthenticationFilter.class)
-        .authorizeRequests().anyRequest().authenticated().and().formLogin().loginPage("/api/login")
-        .successHandler(authenticationSuccessHandler).failureHandler(authenticationFailureHandler)
-        .and().logout().logoutRequestMatcher(new AntPathRequestMatcher("/api/logout"))
-        .logoutSuccessHandler(logoutSuccess).deleteCookies(TOKEN_COOKIE);
-
-  }
+    private final CustomUserDetailsService jwtUserDetailsService;
+    private final RestAuthenticationEntryPoint restAuthenticationEntryPoint;
+    private final LogoutSuccess logoutSuccess;
+    private final AuthenticationSuccessHandler authenticationSuccessHandler;
+    private final AuthenticationFailureHandler authenticationFailureHandler;
+    @Value("${jwt.cookie}")
+    private String TOKEN_COOKIE;
+
+    @Autowired
+    public WebSecurityConfig(@Lazy CustomUserDetailsService jwtUserDetailsService, RestAuthenticationEntryPoint restAuthenticationEntryPoint, LogoutSuccess logoutSuccess, @Lazy AuthenticationSuccessHandler authenticationSuccessHandler, AuthenticationFailureHandler authenticationFailureHandler) {
```

Comment:
Circular dependencies we need to decide how to improve this so that we do not need to use @Lazy

Reply:
I agree. An easy fix is to relocate the `changePassword` method in `CustomUserDetailsService`. What do you think?

Reply:
I did try where would you like the changePassword to go?